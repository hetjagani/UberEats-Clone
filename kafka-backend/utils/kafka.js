const sendData = (producer, data, isError) => {
  producer.send(
    [{ topic: 'response', messages: JSON.stringify({ data, isError }), partitions: 1 }],
    (err, d) => {
      if (err) {
        console.error(err);
        throw Error('error sending data to kafka');
      } else {
        console.log('SENT DATA:', d);
      }
    },
  );
};

module.exports = {
  sendData,
};
