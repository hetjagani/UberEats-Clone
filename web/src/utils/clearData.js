export const clearData = () => {
  document.cookie.split(';').forEach(function (c) {
    document.cookie = c
      .replace(/^ +/, '')
      .replace(/=.*/, '=;expires=' + new Date().toUTCString() + ';path=/');

    window.localStorage.removeItem('persist:root');
  });
};
