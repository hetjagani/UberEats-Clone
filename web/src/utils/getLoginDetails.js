import jwt from 'jsonwebtoken';
import { getCookie } from 'react-use-cookie';

const getLoginDetails = () => {
  const token = getCookie('auth');
  if (token) {
    const tokendata = jwt.decode(token);
    return tokendata;
  } else {
    return {};
  }
};

export default getLoginDetails;
