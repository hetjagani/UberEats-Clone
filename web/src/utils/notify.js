import { toast } from 'react-toastify';

const notify = ({ type, description }) => {
  toast[type](`${type}: ${description}`, {
    position: 'top-center',
    autoClose: 5000,
    hideProgressBar: false,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    progress: undefined,
  });
};

export default notify;
