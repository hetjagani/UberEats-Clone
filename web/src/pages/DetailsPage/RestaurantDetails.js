import { useStyletron } from 'baseui';
import React, { useEffect, useState } from 'react';
import { FormControl } from 'baseui/form-control';
import { Input, MaskedInput } from 'baseui/input';
import { Select } from 'baseui/select';
import { Button, SIZE } from 'baseui/button';
import { Textarea } from 'baseui/textarea';
import { Display4 } from 'baseui/typography';
import { Card } from 'baseui/card';
import {
  clearRestaurantMedia,
  createRestaurant,
  createRestaurantMedia,
  updateRestaurant,
} from '../../actions/restaurants';
import { useDispatch, useSelector } from 'react-redux';
import { FileUploader } from 'baseui/file-uploader';
import S3 from 'react-aws-s3';
import 'react-responsive-carousel/lib/styles/carousel.min.css'; // requires a loader
import { Carousel } from 'react-responsive-carousel';
import notify from '../../utils/notify';
import { Redirect } from 'react-router';

const RestaurantDetails = ({ loginDetails, update }) => {
  const [css, theme] = useStyletron();
  const formContainer = css({
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    width: '80%',
    marginRight: '50px',
  });
  const mainContainer = css({
    display: 'flex',
    justifyContent: 'space-between',
    width: '100%',
  });
  const imgContainer = css({
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    marginLeft: '50px',
    width: '30vw',
  });

  const cityOpts = [
    { id: 'San Jose', city: 'San Jose' },
    { id: 'Berkely', city: 'Berkely' },
    { id: 'Los Angeles', city: 'Los Angeles' },
    { id: 'San Diego', city: 'San Diego' },
    { id: 'San Francisco', city: 'San Francisco' },
    { id: 'Fresno', city: 'Fresno' },
    { id: 'Sacremento', city: 'Sacremento' },
  ];

  const stateOpts = [
    { id: 'California', state: 'California' },
    { id: 'Arizona', state: 'Arizona' },
    { id: 'Colorado', state: 'Colorado' },
  ];

  const countryOpts = [{ id: 'United States', country: 'United States' }];

  const { media, loginRestaurant } = useSelector((state) => {
    return { media: state.restaurants.media, loginRestaurant: state.restaurants.loginRestaurant };
  });

  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const [address, setAddress] = useState('');
  const [city, setCity] = useState([]);
  const [state, setState] = useState([]);
  const [country, setCountry] = useState([]);
  const [contactNo, setContactNo] = useState('');
  const [timeOpen, setTimeOpen] = useState('');
  const [timeClose, setTimeClose] = useState('');
  const [foodType, setFoodType] = useState([]);
  const [resType, setResType] = useState([]);

  useEffect(() => {
    if (update) {
      setName(loginRestaurant.name);
      setDescription(loginRestaurant.description);
      setAddress(loginRestaurant.address);
      setCity([{ city: loginRestaurant.city }]);
      setState([{ state: loginRestaurant.state }]);
      setCountry([{ country: loginRestaurant.country }]);
      setContactNo(loginRestaurant.contact_no);
      setTimeOpen(loginRestaurant.time_open);
      setTimeClose(loginRestaurant.time_close);
      setFoodType([{ id: loginRestaurant.food_type }]);
      setResType([{ id: loginRestaurant.restaurant_type }]);
    }
  }, []);

  const [isUploading, setIsUploading] = useState(false);
  const [isChanged, setIsChanged] = useState(false);

  const dispatch = useDispatch();

  const saveRestaurantDetails = () => {
    if (
      name == '' ||
      description == '' ||
      address == '' ||
      city == [] ||
      state == [] ||
      country == [] ||
      contactNo == '' ||
      timeOpen == '' ||
      timeClose == '' ||
      foodType == [] ||
      resType == []
    ) {
      notify({ type: 'error', description: 'Invalid Values' });
      return;
    }
    const data = {
      id: loginDetails.id,
      name,
      description,
      address,
      city: city[0] && city[0].city,
      state: state[0] && state[0].state,
      country: country[0] && country[0].country,
      contact_no: contactNo,
      time_open: timeOpen,
      time_close: timeClose,
      food_type: foodType[0] && foodType[0].id,
      restaurant_type: resType[0] && resType[0].id,
      media: media,
    };

    if (update) {
      dispatch(updateRestaurant(data, data.id)).then(() => {
        setIsChanged(true);
      });
    } else {
      dispatch(createRestaurant(data)).then(() => {
        setIsChanged(true);
      });
    }
  };

  const mediaUpload = (acceptedFiles, rejectedFiles) => {
    const config = {
      bucketName: window.AWS_BUCKET_NAME,
      region: window.AWS_REGION,
      accessKeyId: window.AWS_ACCESS_ID,
      secretAccessKey: window.AWS_ACCESS_KEY,
      s3Url: 'https://uber-eats-media.s3.amazonaws.com',
    };

    const s3 = new S3(config);

    setIsUploading(true);

    acceptedFiles.forEach((file) => {
      s3.uploadFile(file, file.name)
        .then((res) => {
          notify({ type: 'info', description: `File ${res.key} uploaded...` });

          if (res.status == 204) {
            dispatch(createRestaurantMedia({ alt_text: res.key, url: res.location }));
          }
          setIsUploading(false);
        })
        .catch((err) => {
          console.log(err);
          notify({ type: 'error', description: `Uploading File Failed` });
        });
    });
  };

  const clearMedia = () => {
    dispatch(clearRestaurantMedia());
  };

  if (isChanged) {
    return <Redirect to="/dashboard" />;
  }

  return (
    <div className={mainContainer}>
      <div className={formContainer}>
        <Display4>Restaurant Profile</Display4>
        <FormControl label={() => 'Name'}>
          <Input value={name} onChange={(e) => setName(e.target.value)} required />
        </FormControl>

        <FormControl label="Description" caption="Tell us something about your restaurant">
          <Textarea
            value={description}
            onChange={(event) => setDescription(event.currentTarget.value)}
          />
        </FormControl>

        <FormControl label={() => 'Address'}>
          <Input value={address} onChange={(e) => setAddress(e.target.value)} required />
        </FormControl>

        <FormControl label="Type to search city">
          <Select
            value={city}
            onChange={({ value }) => setCity(value)}
            options={cityOpts}
            labelKey="id"
            valueKey="city"
            required
          />
        </FormControl>

        <FormControl label="Type to search state">
          <Select
            value={state}
            onChange={({ value }) => setState(value)}
            options={stateOpts}
            labelKey="id"
            valueKey="state"
            required
          />
        </FormControl>

        <FormControl label="Type to search country">
          <Select
            value={country}
            onChange={({ value }) => setCountry(value)}
            options={countryOpts}
            labelKey="id"
            valueKey="country"
            required
          />
        </FormControl>

        <FormControl label={() => 'Contact No'}>
          <MaskedInput
            placeholder="Phone number"
            mask="(999) 999-9999"
            value={contactNo}
            onChange={(e) => setContactNo(e.target.value)}
          />
        </FormControl>

        <FormControl label={() => 'Opening Time'}>
          <MaskedInput
            placeholder="Opening Time"
            mask="99:99"
            value={timeOpen}
            onChange={(e) => setTimeOpen(e.target.value)}
          />
        </FormControl>

        <FormControl label={() => 'Closing Time'}>
          <MaskedInput
            placeholder="Closing Time"
            mask="99:99"
            value={timeClose}
            onChange={(e) => setTimeClose(e.target.value)}
          />
        </FormControl>

        <FormControl label="Type of Food">
          <Select
            value={foodType}
            onChange={({ value }) => setFoodType(value)}
            options={[
              { id: 'veg', type: 'Vegitarian' },
              { id: 'non-veg', type: 'Non Vegitarian' },
              { id: 'vegan', type: 'Vegan' },
            ]}
            labelKey="type"
            valueKey="id"
            required
          />
        </FormControl>

        <FormControl label="Type of Restaurant">
          <Select
            value={resType}
            onChange={({ value }) => setResType(value)}
            options={[
              { id: 'delivery', type: 'Delivery' },
              { id: 'pickup', type: 'Pickup' },
            ]}
            labelKey="type"
            valueKey="id"
            searchable={false}
            required
          />
        </FormControl>

        <Button
          type="submit"
          $style={{ alignSelf: 'baseline', width: '30%' }}
          size={SIZE.large}
          onClick={saveRestaurantDetails}
        >
          Save
        </Button>
      </div>

      <div className={imgContainer}>
        <Carousel $style={{ width: '80%' }}>
          {media &&
            media.length > 0 &&
            media.map((m) => {
              return (
                <div key={m.id}>
                  <img src={m.url} alt={m.alt_text} width="500" />
                </div>
              );
            })}
        </Carousel>
        <Button className={css({ margin: '20px' })} onClick={clearMedia}>
          Clear Images
        </Button>
        <FileUploader
          onCancel={() => setIsUploading(false)}
          onDrop={mediaUpload}
          progressMessage={isUploading ? `Uploading... hang tight.` : ''}
        />
      </div>
    </div>
  );
};

export default RestaurantDetails;
