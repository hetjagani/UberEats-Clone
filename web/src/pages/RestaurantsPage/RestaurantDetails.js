import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router';
import 'react-responsive-carousel/lib/styles/carousel.min.css'; // requires a loader
import './carousel_override.css';
import { Carousel } from 'react-responsive-carousel';
import { useStyletron } from 'baseui';
import { H5, Paragraph1, Paragraph2 } from 'baseui/typography';
import { FlexGrid, FlexGridItem } from 'baseui/flex-grid';
import axios from 'axios';
import notify from '../../utils/notify';
import NavBar from './NavBar';
import DishDetails from './DishDetails';
import withAuth from '../AuthPage/withAuth';
import { Modal, ModalHeader, ModalBody, ModalFooter, ModalButton } from 'baseui/modal';
import { useDispatch, useSelector } from 'react-redux';
import { clearCart } from '../../actions/cart';
import query from '../../utils/graphql/query';
import { restaurantQuery } from '../../queries/queries';

const RestaurantDetails = () => {
  const { id } = useParams();

  const [css] = useStyletron();
  const imgContainer = css({
    display: 'flex',
    justifyContent: 'center',
    marginTop: '30px',
  });

  const mainContainer = css({
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    width: '100vw',
  });

  const itemProps = {
    height: 'auto',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  };

  const [restaurant, setRestaurant] = useState({});
  const [media, setMedia] = useState({});
  const [dishes, setDishes] = useState({});
  const [errDiffRes, setErrDiffRes] = useState(false);

  const { diffRestaurantError } = useSelector((state) => {
    return { diffRestaurantError: state.cart.diffRestaurantError };
  });

  const dispatch = useDispatch();
  const resetCart = () => {
    dispatch(clearCart());
    setErrDiffRes(false);
  };

  useEffect(() => {
    setErrDiffRes(true);
  }, [diffRestaurantError.time]);

  const closeErrDiff = () => {
    setErrDiffRes(false);
  };

  useEffect(() => {
    setErrDiffRes(false);
    const variables = { id };
    query(restaurantQuery, variables)
      .then((res) => {
        setRestaurant(res.restaurant);
        setMedia(res.restaurant.media);
        setDishes(res.restaurant.dishes);
      })
      .catch((err) => {
        console.error(err);
        notify({ type: 'error', description: 'Error fetching restaurant' });
      });
  }, []);

  return (
    <div style={{ display: 'flex', alignItems: 'center', flexDirection: 'column' }}>
      <div className={mainContainer}>
        <NavBar />
        <div className={imgContainer}>
          <Carousel
            autoPlay={true}
            dynamicHeight={true}
            interval={5000}
            showThumbs={false}
            infiniteLoop={true}
            showIndicators={false}
          >
            {media &&
              media.length > 0 &&
              media.map((m) => {
                return (
                  <div key={m.id}>
                    <img src={m.url} alt={m.alt_text} width="500" />
                  </div>
                );
              })}
          </Carousel>
        </div>

        <div
          className={css({
            alignSelf: 'baseline',
            margin: '30px',
            display: 'flex',
            justifyContent: 'space-between',
          })}
        >
          <div className={css({ width: '80vw', marginLeft: '130px' })}>
            <H5>
              {restaurant.name} (
              {restaurant.restaurant_type && restaurant.restaurant_type.toUpperCase()})
            </H5>
            <Paragraph2>
              <strong>Description: </strong>
              {restaurant.description}
            </Paragraph2>
            <Paragraph2>
              <strong>Address: </strong>
              {restaurant.address}
            </Paragraph2>
            <Paragraph2>
              <strong>Location: </strong>
              {restaurant.city} | {restaurant.state} | {restaurant.country}
            </Paragraph2>
          </div>
        </div>
      </div>
      <div style={{ width: '90vw', margin: '20px' }}>
        <FlexGrid flexGridColumnCount={3} flexGridColumnGap="scale800" flexGridRowGap="scale800">
          {dishes.length > 0 &&
            dishes.map((dish) => {
              return (
                <FlexGridItem {...itemProps} key={dish._id}>
                  <DishDetails dish={dish} resId={restaurant._id} />
                </FlexGridItem>
              );
            })}
        </FlexGrid>
      </div>

      <Modal
        onClose={closeErrDiff}
        isOpen={errDiffRes}
        overrides={{
          Dialog: {
            style: {
              width: '40vw',
              height: 'fit-content',
              display: 'flex',
              flexDirection: 'column',
            },
          },
        }}
      >
        <ModalHeader>Trying to add dish from other restaurant</ModalHeader>
        <ModalBody style={{ flex: '1 1 0' }}>
          <Paragraph1>
            {' '}
            <div dangerouslySetInnerHTML={{ __html: diffRestaurantError.message }}></div>
          </Paragraph1>
          <Paragraph1>You need to clear cart in order to add dish from this restaurant.</Paragraph1>
        </ModalBody>
        <ModalFooter>
          <ModalButton kind="tertiary" onClick={closeErrDiff}>
            Cancel
          </ModalButton>
          <ModalButton onClick={resetCart}>Reset Cart</ModalButton>
        </ModalFooter>
      </Modal>
    </div>
  );
};

export default withAuth(RestaurantDetails, 'customer');
