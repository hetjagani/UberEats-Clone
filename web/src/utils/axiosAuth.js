import axios from 'axios';
import { getCookie, setCookie } from 'react-use-cookie';
import notify from './notify';

function createAxiosAuthMiddleware() {
  return ({ getState }) =>
    (next) =>
    (action) => {
      const authToken = getCookie('auth');

      if (authToken) {
        axios.defaults.headers.common['Authorization'] = authToken;
      }

      axios.defaults.baseURL = window.BACKEND_API_URL;
      axios.defaults.withCredentials = false;
      axios.interceptors.response.use(
        (res) => res,
        (err) => {
          if (err.response.status === 401) {
            setCookie('auth', null);
            window.location = '/auth/login';
            notify({ type: 'error', description: 'Unauthorized. Please Login.' });
          }
          return Promise.reject(err);
        },
      );
      return next(action);
    };
}

export default createAxiosAuthMiddleware;
