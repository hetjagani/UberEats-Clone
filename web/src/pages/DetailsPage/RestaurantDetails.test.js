
import { shallow } from 'enzyme';
import React from 'react';
import '../../utils/testsetup';
import { Display4 } from 'baseui/typography';
import RestaurantDetails from './RestaurantDetails';

import * as redux from 'react-redux';

describe('Restaurant Details Page Test', () => {
  const initialState = {
    medium: [],
    loginRestaurant: {
      name: 'Test User',
      description: 'Test About',
      city: 'Fremont',
      state: 'California',
      country: 'USA',
      contact_no: '9898989898',
    },
  };

  const spy = jest.spyOn(redux, 'useSelector');
  spy.mockReturnValue(initialState);
  const spy2 = jest.spyOn(redux, 'useDispatch');
  spy2.mockReturnValue(null);

  it('should render Restaurant Details page', () => {
    const detailsPage = shallow(<RestaurantDetails />);
    expect(detailsPage.getElements()).toMatchSnapshot();
  });

  it('should have Restaurant Profile heading', () => {
    const detailsPage = shallow(<RestaurantDetails />);
    expect(detailsPage.find(Display4).text()).toBe('Restaurant Profile');
  });
});
