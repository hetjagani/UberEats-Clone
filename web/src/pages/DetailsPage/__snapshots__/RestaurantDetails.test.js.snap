// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant Details Page Test should render Restaurant Details page 1`] = `
Array [
  <div
    className=""
  >
    <div
      className=""
    >
      <DisplayXSmall>
        Restaurant Profile
      </DisplayXSmall>
      <FormControl
        caption={null}
        disabled={false}
        label={[Function]}
        overrides={Object {}}
      >
        <Input
          autoComplete="on"
          autoFocus={false}
          clearable={false}
          disabled={false}
          endEnhancer={null}
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          overrides={Object {}}
          required={true}
          size="default"
          startEnhancer={null}
          type="text"
          value=""
        />
      </FormControl>
      <FormControl
        caption="Tell us something about your restaurant"
        disabled={false}
        label="Description"
        overrides={Object {}}
      >
        <Textarea
          autoFocus={false}
          disabled={false}
          error={false}
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          overrides={Object {}}
          placeholder=""
          required={false}
          rows={3}
          size="default"
          value=""
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label={[Function]}
        overrides={Object {}}
      >
        <Input
          autoComplete="on"
          autoFocus={false}
          clearable={false}
          disabled={false}
          endEnhancer={null}
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          overrides={Object {}}
          required={true}
          size="default"
          startEnhancer={null}
          type="text"
          value=""
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label="Type to search city"
        overrides={Object {}}
      >
        <Select
          labelKey="id"
          onChange={[Function]}
          options={
            Array [
              Object {
                "city": "San Jose",
                "id": "San Jose",
              },
              Object {
                "city": "Berkely",
                "id": "Berkely",
              },
              Object {
                "city": "Los Angeles",
                "id": "Los Angeles",
              },
              Object {
                "city": "San Diego",
                "id": "San Diego",
              },
              Object {
                "city": "San Francisco",
                "id": "San Francisco",
              },
              Object {
                "city": "Fresno",
                "id": "Fresno",
              },
              Object {
                "city": "Sacremento",
                "id": "Sacremento",
              },
            ]
          }
          required={true}
          value={Array []}
          valueKey="city"
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label="Type to search state"
        overrides={Object {}}
      >
        <Select
          labelKey="id"
          onChange={[Function]}
          options={
            Array [
              Object {
                "id": "California",
                "state": "California",
              },
              Object {
                "id": "Arizona",
                "state": "Arizona",
              },
              Object {
                "id": "Colorado",
                "state": "Colorado",
              },
            ]
          }
          required={true}
          value={Array []}
          valueKey="state"
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label="Type to search country"
        overrides={Object {}}
      >
        <Select
          labelKey="id"
          onChange={[Function]}
          options={
            Array [
              Object {
                "country": "United States",
                "id": "United States",
              },
            ]
          }
          required={true}
          value={Array []}
          valueKey="country"
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label={[Function]}
        overrides={Object {}}
      >
        <MaskedInput
          mask="(999) 999-9999"
          maskChar=" "
          onChange={[Function]}
          placeholder="Phone number"
          value=""
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label={[Function]}
        overrides={Object {}}
      >
        <MaskedInput
          mask="99:99"
          maskChar=" "
          onChange={[Function]}
          placeholder="Opening Time"
          value=""
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label={[Function]}
        overrides={Object {}}
      >
        <MaskedInput
          mask="99:99"
          maskChar=" "
          onChange={[Function]}
          placeholder="Closing Time"
          value=""
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label="Type of Food"
        overrides={Object {}}
      >
        <Select
          labelKey="type"
          onChange={[Function]}
          options={
            Array [
              Object {
                "id": "veg",
                "type": "Vegitarian",
              },
              Object {
                "id": "non-veg",
                "type": "Non Vegitarian",
              },
              Object {
                "id": "vegan",
                "type": "Vegan",
              },
            ]
          }
          required={true}
          value={Array []}
          valueKey="id"
        />
      </FormControl>
      <FormControl
        caption={null}
        disabled={false}
        label="Type of Restaurant"
        overrides={Object {}}
      >
        <Select
          labelKey="type"
          onChange={[Function]}
          options={
            Array [
              Object {
                "id": "delivery",
                "type": "Delivery",
              },
              Object {
                "id": "pickup",
                "type": "Pickup",
              },
            ]
          }
          required={true}
          searchable={false}
          value={Array []}
          valueKey="id"
        />
      </FormControl>
      <Button
        $style={
          Object {
            "alignSelf": "baseline",
            "width": "30%",
          }
        }
        onClick={[Function]}
        size="large"
        type="submit"
      >
        Save
      </Button>
    </div>
    <div
      className=""
    >
      <Carousel
        $style={
          Object {
            "width": "80%",
          }
        }
        axis="horizontal"
        centerSlidePercentage={80}
        interval={3000}
        labels={
          Object {
            "item": "slide item",
            "leftArrow": "previous slide / item",
            "rightArrow": "next slide / item",
          }
        }
        onChange={[Function]}
        onClickItem={[Function]}
        onClickThumb={[Function]}
        onSwipeEnd={[Function]}
        onSwipeMove={[Function]}
        onSwipeStart={[Function]}
        preventMovementUntilSwipeScrollTolerance={false}
        renderArrowNext={[Function]}
        renderArrowPrev={[Function]}
        renderIndicator={[Function]}
        renderItem={[Function]}
        renderThumbs={[Function]}
        selectedItem={0}
        showArrows={true}
        showIndicators={true}
        showStatus={true}
        showThumbs={true}
        statusFormatter={[Function]}
        stopOnHover={true}
        swipeScrollTolerance={5}
        swipeable={true}
        transitionTime={350}
        verticalSwipe="standard"
        width="100%"
      />
      <Button
        className=""
        onClick={[Function]}
      >
        Clear Images
      </Button>
      <FileUploader
        disableClick={true}
        onCancel={[Function]}
        onDrop={[Function]}
        overrides={Object {}}
        progressMessage=""
      />
    </div>
  </div>,
]
`;
